sdk/core/AzureCore/Tests/QueryParametersTests.swift

```diff
@@ -0,0 +1,168 @@
+// --------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation. All rights reserved.
+//
+// The MIT License (MIT)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the ""Software""), to
+// deal in the Software without restriction, including without limitation the
+// rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+// sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//
+// --------------------------------------------------------------------------
+
+@testable import AzureCore
+import XCTest
+
+// MARK: Test Object Defintions
+
+enum AnimalEnum: String, QueryStringConvertible {
+    case cat
+    case dog
+
+    var queryValue: String {
+        return rawValue
+    }
+}
+
+enum FlavorEnum: Int, QueryStringConvertible {
+    case chocolate
+    case vanilla
+
+    var queryValue: String {
+        return String(rawValue)
+    }
+}
+
+enum ShapeEnum: QueryStringConvertible {
+    case custom(String)
+    case square
+    case circle
+
+    var queryValue: String {
+        switch self {
+        case let .custom(val):
+            return val
+        case .square:
+            return "square"
+        case .circle:
+            return "circle"
+        }
+    }
+}
+
+typealias MyDate = Date
+extension MyDate: QueryStringConvertible {
+    internal static let dateFormat = "yyyy-MM-dd"
+
+    public var queryValue: String {
+        let formatter = DateFormatter()
+        formatter.dateFormat = MyDate.dateFormat
+        return formatter.string(from: self)
+    }
+}
```

Comment:
While we can put a generic implementation into AzureCore, I don't think it will work, generally. @sacheu I think we'd need to actually generate the Date schemas from Swagger in a manner similar to this, where each one can specify its own format. Curious what you think.

Reply:
This solution didn't work and has been replaced.